evm/src/main/java/org/hyperledger/besu/evm/EVM.java

```diff
@@ -239,10 +239,10 @@ public void runToHalt(final MessageFrame frame, final OperationTracer tracing) {
               case 0x03 -> SubOperation.staticOperation(frame);
               case 0x04 -> DivOperation.staticOperation(frame);
               case 0x05 -> SDivOperation.staticOperation(frame);
-              case 0x06 -> ModOperation.staticOperation(frame);
-              case 0x07 -> SModOperation.staticOperation(frame);
-              case 0x08 -> AddModOperation.staticOperation(frame);
-              case 0x09 -> MulModOperation.staticOperation(frame);
```

Comment:
As you're doing a static binding here, there optimized implementations will always be executed. You need to get them from the OperationRegistery that is built following the EVM configuration.